thetis/solver.py

```diff
@@ -618,3 +650,4 @@ def iterate(self, update_forcings=None, update_forcings3d=None,
                         print '{0:25s} : {1:11.6f} {2:11.2f}'.format(
                             label, c, relcost)
                         sys.stdout.flush()
+                gc.collect()
```

Comment:
Yes, simulations eventually die on stampede as the nodes run out of memory. AFAIK invoking garbage collector manually doesn't really help. At least now I'm able to restart the interrupted runs.